Broken Unit Tests

### Describe the bug

When trying to run npm run test, we have the following compilation errors:

Error: src/app/components/picklist/picklist.spec.ts:253:18 - error TS2554: Expected 5-6 arguments, but got 4.

253         picklist.onItemClick(event, picklist.source[0], picklist.selectedItemsSource, callback);
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/components/picklist/picklist.ts:922:88
    922     onItemClick(event: Event | any, item: any, selectedItems: any[], listType: number, callback: EventEmitter<any>, itemId?: string) {
                                                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'callback' was not provided.



\ Generating browser application bundles (phase: building)...01 05 2024 08:35:45.785:WARN [karma]: No captured browser, open http://localhost:9877/
01 05 2024 08:35:45.861:INFO [karma-server]: Karma v6.4.3 server started at http://localhost:9877/
01 05 2024 08:35:45.862:INFO [launcher]: Launching browsers Chrome with concurrency unlimited
01 05 2024 08:35:45.874:INFO [launcher]: Starting browser Chrome
√ Browser application bundle generation complete.

Error: src/app/components/picklist/picklist.spec.ts:253:18 - error TS2554: Expected 5-6 arguments, but got 4.

253         picklist.onItemClick(event, picklist.source[0], picklist.selectedItemsSource, callback);
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/components/picklist/picklist.ts:922:88
    922     onItemClick(event: Event | any, item: any, selectedItems: any[], listType: number, callback: EventEmitter<any>, itemId?: string) {
                                                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'callback' was not provided.



01 05 2024 08:35:47.791:INFO [Chrome 123.0.0.0 (Windows 10)]: Connected on socket qxY6PvR5UzstF3J0AAAB with id 89045588
01 05 2024 08:35:48.844:WARN [web-server]: 404: /_karma_webpack_/main.js
Chrome 123.0.0.0 (Windows 10): Executed 0 of 0 SUCCESS (0.002 secs / 0 secs)
TOTAL: 0 SUCCESS



### Environment

windows


### Reproducer

_No response_

### Angular version

17.3.6

### PrimeNG version

17.15.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v21.7.3

### Browser(s)

_No response_

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

--- a/src/app/components/checkbox/checkbox.spec.ts
+++ b/src/app/components/checkbox/checkbox.spec.ts
@@ -123,8 +123,8 @@ describe('Checkbox', () => {
         const input = fixture.nativeElement.querySelector('input');
         const boxEl = fixture.nativeElement.querySelector('.p-checkbox-box');
         const labelEl = fixture.nativeElement.querySelector('.p-checkbox-label');
-        const onBlurSpy = spyOn(checkbox, 'onBlur').and.callThrough();
-        const onFocusSpy = spyOn(checkbox, 'onFocus').and.callThrough();
+        const onBlurSpy = spyOn(checkbox, 'onInputBlur').and.callThrough();
+        const onFocusSpy = spyOn(checkbox, 'onInputFocus').and.callThrough();
         input.dispatchEvent(new Event('focus'));
         fixture.detectChanges();
 

--- a/src/app/components/menubar/menubar.spec.ts
+++ b/src/app/components/menubar/menubar.spec.ts
@@ -1,16 +1,16 @@
 import { ComponentFixture, fakeAsync, flush, TestBed, tick } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-import { RouterTestingModule } from '@angular/router/testing';
 import { Menubar, MenubarModule, MenubarService, MenubarSub } from './menubar';
+import { RouterModule } from '@angular/router';
 
-describe('Menubar', () => {
+fdescribe('Menubar', () => {
     let menubar: Menubar;
     let fixture: ComponentFixture<Menubar>;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [NoopAnimationsModule, RouterTestingModule.withRoutes([{ path: 'test', component: Menubar }]), MenubarModule],
+            imports: [NoopAnimationsModule, RouterModule.forRoot([{ path: 'test', component: Menubar }]), MenubarModule],
             providers: [MenubarService]
         });
 
@@ -45,7 +45,7 @@ describe('Menubar', () => {
         const subMenu = fixture.debugElement.query(By.css('.p-menubar-root-list')).componentInstance as MenubarSub;
         expect(subMenu.baseZIndex).toEqual(20);
         expect(subMenu.autoZIndex).toEqual(false);
-        expect(subMenu.autoDisplay).toBeUndefined();
+        expect(subMenu.autoDisplay).toBeDefined();
         expect(subMenu.autoDisplay).toEqual(menubar.autoDisplay);
         expect(subMenu.autoZIndex).toEqual(menubar.autoZIndex);
         expect(subMenu.baseZIndex).toEqual(menubar.baseZIndex);
@@ -121,53 +121,11 @@ describe('Menubar', () => {
         const parentEl = fixture.debugElement.query(By.css('.p-menubar-root-list'));
         const firstParentEl = parentEl.query(By.css('.p-menuitem-link'));
         const secondParentEl = parentEl.queryAll(By.css('.p-menuitem-link'))[1];
-        firstParentEl.nativeElement.dispatchEvent(new Event('click'));
+        firstParentEl.nativeElement.click();
         fixture.detectChanges();
 
-        expect(firstParentEl.parent.nativeElement.classList.contains('p-menuitem-active')).toBe(true);
-        expect(secondParentEl.parent.nativeElement.classList.contains('p-menuitem-active')).toBe(false);
-    });
-
-    it('should call itemClick', () => {
-        menubar.model = [
-            {
-                label: 'File',
-                icon: 'pi pi-fw pi-file',
-                items: [
-                    {
-                        label: 'New',
-                        icon: 'pi pi-fw pi-plus'
-                    },
-                    { label: 'Open' },
-                    { separator: true },
-                    { label: 'Quit' }
-                ]
-            },
-            {
-                label: 'Edit',
-                icon: 'pi pi-fw pi-pencil',
-                items: [
-                    { label: 'Delete', icon: 'pi pi-fw pi-trash' },
-                    { label: 'Refresh', icon: 'pi pi-fw pi-refresh' }
-                ]
-            }
-        ];
-        fixture.detectChanges();
-
-        const parentEl = fixture.debugElement.query(By.css('.p-menubar-root-list'));
-        const firstParentEl = parentEl.query(By.css('.p-menuitem-link'));
-        const secondParentEl = parentEl.queryAll(By.css('.p-menuitem-link'))[1];
-        firstParentEl.nativeElement.dispatchEvent(new Event('click'));
-        fixture.detectChanges();
-
-        const firstSubmenuList = fixture.debugElement.query(By.css('.p-submenu-list'));
-        const firstSubItem = firstSubmenuList.query(By.css('.p-menuitem-link'));
-        firstSubItem.nativeElement.click();
-        fixture.detectChanges();
-
-        expect(firstParentEl.componentInstance.activeItem).toEqual(null);
-        expect(secondParentEl.componentInstance.activeItem).toEqual(null);
-        expect(firstParentEl.parent.nativeElement.className).not.toContain('p-menuitem-active');
+        expect(firstParentEl?.parent?.parent?.nativeElement.classList.contains('p-menuitem-active')).toBe(true);
+        expect(secondParentEl?.parent?.parent?.nativeElement.classList.contains('p-menuitem-active')).toBe(false);
     });
 
     it('should call onItemMouseEnter and not show firstParentMenu', () => {
@@ -241,11 +199,11 @@ describe('Menubar', () => {
         firstParentEl.nativeElement.dispatchEvent(new Event('mouseleave'));
         tick(300);
         fixture.detectChanges();
-        expect(firstParentEl.parent.nativeElement.classList.contains('p-menuitem-active')).toBe(true);
+        expect(firstParentEl?.parent?.parent?.nativeElement.classList.contains('p-menuitem-active')).toBe(true);
         flush();
     }));
 
-    it('should call itemClick and bindEventListener', () => {
+    it('should call itemClick', () => {
         menubar.model = [
             {
                 label: 'File',
@@ -273,7 +231,7 @@ describe('Menubar', () => {
 
         const parentEl = fixture.debugElement.query(By.css('.p-menubar-root-list'));
         const firstParentEl = parentEl.query(By.css('.p-menuitem-link'));
-        const bindEventListenerSpy = spyOn(firstParentEl.componentInstance, 'bindDocumentClickListener').and.callThrough();
+        const secondParentEl = parentEl.queryAll(By.css('.p-menuitem-link'))[1];
         firstParentEl.nativeElement.click();
         fixture.detectChanges();
 
@@ -282,8 +240,9 @@ describe('Menubar', () => {
         firstSubItem.nativeElement.dispatchEvent(new Event('mouseenter'));
         fixture.detectChanges();
 
-        expect(bindEventListenerSpy).toHaveBeenCalled();
-        expect(firstParentEl.parent.nativeElement.className).toContain('p-menuitem-active');
+        expect(firstParentEl?.parent?.parent?.nativeElement.className).toContain('p-menuitem-active');
+        expect(secondParentEl?.parent?.parent?.nativeElement.className).not.toContain('p-menuitem-active');
+
     });
 
     it('should show router items', () => {
@@ -315,10 +274,9 @@ describe('Menubar', () => {
 
         const parentEl = fixture.debugElement.query(By.css('.p-menubar-root-list'));
         const firstParentEl = parentEl.query(By.css('.p-menuitem-link'));
-        firstParentEl.nativeElement.dispatchEvent(new Event('click'));
+        firstParentEl.nativeElement.click();
         fixture.detectChanges();
-
-        expect(firstParentEl.componentInstance.activeItem.label).toEqual(firstParentEl.nativeElement.textContent);
+        expect(firstParentEl?.componentInstance.activeItemPath[0].item.label).toEqual(firstParentEl.nativeElement.textContent);
     });
 
     it('should call itemClick', () => {

--- a/src/app/components/menubar/menubar.spec.ts
+++ b/src/app/components/menubar/menubar.spec.ts
@@ -4,7 +4,7 @@ import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { Menubar, MenubarModule, MenubarService, MenubarSub } from './menubar';
 import { RouterModule } from '@angular/router';
 
-fdescribe('Menubar', () => {
+describe('Menubar', () => {
     let menubar: Menubar;
     let fixture: ComponentFixture<Menubar>;
 

--- a/src/app/components/picklist/picklist.spec.ts
+++ b/src/app/components/picklist/picklist.spec.ts
@@ -1,6 +1,6 @@
 import { TestBed, ComponentFixture } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
-import { PickListModule, PickList } from 'primeng/picklist';
+import { PickListModule, PickList } from './picklist';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { Component, EventEmitter } from '@angular/core';
 
@@ -106,6 +106,7 @@ describe('PickList', () => {
         fixture.detectChanges();
 
         const sourceListItems = fixture.debugElement.query(By.css('.p-picklist-source-wrapper')).queryAll(By.css('.p-picklist-item'));
+       //expect(picklist.visibleOptionsSource).toBeDefined();
         expect(picklist.visibleOptionsSource.length).toEqual(2);
         expect(picklist.visibleOptionsSource[0].brand).toEqual('VW');
         expect(picklist.visibleOptionsSource[1].brand).toEqual('Volvo');
@@ -250,7 +251,7 @@ describe('PickList', () => {
 
         let event = { ctrlKey: true };
         let callback = new EventEmitter();
-        picklist.onItemClick(event, picklist.source[0], picklist.selectedItemsSource, callback);
+        picklist.onItemClick(event, picklist.source, picklist.selectedItemsSource, picklist.SOURCE_LIST, callback);
         fixture.detectChanges();
 
         picklist.cd.detectChanges();

--- a/src/app/components/progressbar/progressbar.spec.ts
+++ b/src/app/components/progressbar/progressbar.spec.ts
@@ -27,7 +27,7 @@ describe('ProgressBar', () => {
         expect(progressbarLabelEl.textContent).toEqual('50%');
     });
 
-    it('should not show value', () => {
+    it('should not show value when showValue is set to false', () => {
         progressbar.value = 50;
         progressbar.showValue = false;
         fixture.detectChanges();
@@ -35,7 +35,7 @@ describe('ProgressBar', () => {
         const progressbarValueEl = fixture.debugElement.query(By.css('.p-progressbar-value')).nativeElement;
         const progressbarLabelEl = fixture.debugElement.query(By.css('.p-progressbar-label'));
         expect(progressbarValueEl.style.width).toEqual('50%');
-        expect(progressbarLabelEl).toBeFalsy();
+        expect(progressbarLabelEl.children.length).toBe(0);
     });
 
     it('should change style and styleClass', () => {

